src/app/core/services/label.service.ts

```diff
@@ -22,17 +27,59 @@ const COLOR_WHITE = 'ffffff'; // Light color for text with dark background
  * from the GitHub repository for the WATcher application.
  */
 export class LabelService {
+  static readonly POLL_INTERVAL = 5000; // 5 seconds
+
   labels: Label[];
+  simplifiedLabels: simplifiedLabel[];
+
+  private labelsPollSubscription: Subscription;
+  private labelsSubject = new BehaviorSubject<simplifiedLabel[]>([]);
 
   constructor(private githubService: GithubService) {}
 
+  startPollLabels() {
+    if (this.labelsPollSubscription === undefined) {
+      this.labelsPollSubscription = timer(0, LabelService.POLL_INTERVAL)
+        .pipe(
+          exhaustMap(() => {
+            return this.fetchLabels().pipe(
+              catchError(() => {
+                return EMPTY;
+              })
+            );
+          })
+        )
+        .subscribe(() => {
+          this.labelsSubject.next(this.simplifiedLabels);
+        });
+    }
+  }
+
+  stopPollLabels() {
+    if (this.labelsPollSubscription) {
+      this.labelsPollSubscription.unsubscribe();
+      this.labelsPollSubscription = undefined;
+    }
+  }
+
+  connect(): Observable<simplifiedLabel[]> {
+    return this.labelsSubject.asObservable();
+  }
+
   /**
    * Fetch labels from Github.
    */
   public fetchLabels(): Observable<any> {
     return this.githubService.fetchAllLabels().pipe(
       map((response) => {
         this.labels = this.parseLabelData(response);
+        this.simplifiedLabels = this.labels.map((label) => {
+          return {
+            name: label.getFormattedName(),
+            color: label.color
+          };
+        });
+        this.labelsSubject.next(this.simplifiedLabels);
```

Comment:
I think at this point it is probably more efficient to just store getFormattedName()'s result as a property of Label instead of cloning the entire label. 

We can still keep simplifiedLabel but make Label implement the type simplified label so this conversion will no longer be needed.

Reply:
Not sure if I implemented the change requested here in the manner that was intended, but I made an attempt to do so, please let me know if I should make further changes